ui-backend/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/cql/SourceWarningsFilterManagerTest.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.cql;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import ddf.catalog.operation.ProcessingDetails;
+import ddf.catalog.security.SourceWarningsFilter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Test;
+
+public class SourceWarningsFilterManagerTest {
+
+  @Test
+  public void testNullProcessingDetailsProduceProcessingDetailsWithEmptyWarnings() {
+    SourceWarningsFilter mockedFilter = mock(SourceWarningsFilter.class);
+    when(mockedFilter.canFilter(any())).thenReturn(true);
+
+    ProcessingDetails mockedDetailsWithNonEmptyWarnings = mock(ProcessingDetails.class);
+    when(mockedDetailsWithNonEmptyWarnings.getWarnings())
+        .thenReturn(Collections.singletonList("doesn't look empty to me, boss"));
+    when(mockedFilter.filter(any())).thenReturn(mockedDetailsWithNonEmptyWarnings);
+
+    SourceWarningsFilterManager filterManager =
+        new SourceWarningsFilterManager(Collections.singletonList(mockedFilter));
+
+    ProcessingDetails detailsWithFilteredWarnings = filterManager.filterWarningsOf(null);
+
+    assertThat(detailsWithFilteredWarnings, is(notNullValue()));
+
+    assertThat(detailsWithFilteredWarnings.getWarnings(), is(Collections.emptyList()));
+  }
+
+  @Test
+  public void testEmptyFiltersProduceProcessingDetailsWithEmptyWarnings() {
+    SourceWarningsFilterManager filterManager =
+        new SourceWarningsFilterManager(Collections.emptyList());
+
+    ProcessingDetails mockedDetailsWithSourceId = mock(ProcessingDetails.class);
+    when(mockedDetailsWithSourceId.getSourceId()).thenReturn("source id");
```

Comment:
✏️ Testing constants can be moved up higher in scope like so:
```
public class SourceWarningsFilterManagerTest {

private static final String FAKE_SOURCE_ID = "source id";
private static final String FAKE_OTHER_THING = "...";

...
```
Note the common prefix of `FAKE_` to benefit from simpler auto-complete.